Makefile

```diff
@@ -339,7 +339,7 @@ endif
 $(GOPATH)/bin/swagger:
 # update this in Nix when upgrading it here
 ifneq ($(USE_NIX), true)
-	go install github.com/go-swagger/go-swagger/cmd/swagger@v0.28.0
+	go install github.com/go-swagger/go-swagger/cmd/swagger@v0.31.0
```

Comment:
@agilgur5 the swagger update as changed/deleted some files which seems unexpected.

Reply:
Reverting to 0.30.5 seems to make this better. I can't see [which change](https://github.com/go-swagger/go-swagger/releases/tag/v0.31.0) would cause this at a quick glance.

Reply:
https://github.com/go-swagger/go-swagger/issues/3070 seems to mention go1.22 might not be compatible with swagger 0.30.5?

Reply:
Yes the go-swagger change is required on go1.22

Reply:
> Reverting to 0.30.5 seems to make this better. I can't see [which change](https://github.com/go-swagger/go-swagger/releases/tag/v0.31.0) would cause this at a quick glance.

Since it only deleted parts of the Swagger, I imagine those parts might have been redundant or unnecessary, so a bug fix could've fixed the redundant/unnecessary codegen. So we just need to review that the deletion is indeed redundant or otherwise unnecessary

Reply:
> I imagine those parts might have been redundant or unnecessary, so a bug fix could've fixed the redundant/unnecessary codegen

Indeed, that looks to be the case, see my [below review comments](https://github.com/argoproj/argo-workflows/pull/13258#pullrequestreview-2172949145)